scripts/validate.sh

```diff
@@ -0,0 +1,73 @@
+#!/bin/bash -e
+
+# Copyright 2018 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# "---------------------------------------------------------"
+# "-                                                       -"
+# "-  Validation script checks if demo application         -"
+# "-  deployed successfully.                               -"
+# "-                                                       -"
+# "---------------------------------------------------------"
+
+# Do no set exit on error, since the rollout status command may fail
+set -o nounset
+set -o pipefail
+
+ROOT=$(dirname "${BASH_SOURCE[0]}")/..
+# shellcheck disable=SC1090
+source "$ROOT"/scripts/common.sh
+
+APP_MESSAGE="deployment \"$APP_NAME\" successfully rolled out"
+
+# Get credentials for the k8s cluster
+gcloud container clusters get-credentials "$CLUSTER_NAME" --zone "$ZONE"
+
+# Wait for the rollout of demo app to finish
+while true
+do
+  ROLLOUT=$(kubectl rollout status --namespace default --watch=false deployment/"$APP_NAME") \
```

Comment:
This could loop forever if you don't get APP_MESSAGE. If it errors out then you're just stuck in a loop waiting.

Reply:
I experienced that very thing. Will work with Victor to find a solution.

Reply:
Just set x number of times that you will wait, and after those x times fail.

Reply:
I was hoping for a better option, but it's probably the best we can do.

Reply:
You are still looping forever.  You need a countdown in here that breaks the loop if you do it 1000 times or something